### PR TITLE
clock: add help prefix to output examples

### LINKED ISSUE #1793: clock: some output lacks help prefix
Examples:

https://github.com/sopel-irc/sopel/blob/ff9ef5ab3bc11cfeba3533e7d3e19e536a08dcfe/sopel/modules/clock.py#L199-L201

https://github.com/sopel-irc/sopel/blob/ff9ef5ab3bc11cfeba3533e7d3e19e536a08dcfe/sopel/modules/clock.py#L302-L305

### PR DESCRIPTION
**Pull request by Exirel**
Closes #1793

### Description

* Add help prefix in output examples, as required by #1793
* Improve and tweak help examples and docstrings

### Checklist
- [x] I have read [CONTRIBUTING.md](https://github.com/sopel-irc/sopel/blob/master/CONTRIBUTING.md)
- [x] I can and do license this contribution under the EFLv2
- [x] No issues are reported by `make qa` (runs `make quality` and `make test`)
- [x] I have tested the functionality of the things this change touches

### CODE REVIEW

File: sopel/modules/clock.py [hunk 1]
```diff
@@ -250,11 +258,12 @@ def update_channel(bot, trigger):
...
-@module.example('.getctz [channel]')
+@module.example('.getctz #sopel', user_help=True)
+@module.example('.getctz', user_help=True)
 def get_channel_tz(bot, trigger):
-    """
-    Gets the channel's preferred timezone; returns the current channel's
-    if no channel name is given.
+    """Gets the channel's preferred timezone.
+
+    It uses the current channel's if no channel name is given.
```
Comment (dgw) [inline sopel/modules/clock.py:266]:
```suggestion
    It uses the current channel if no channel name is given.
```

OR, for consistency with `setctf`:

```suggestion
    It returns the current channel's if no channel name is given.
```

File: sopel/modules/clock.py [hunk 2]
```diff
@@ -33,11 +34,7 @@ def f_time(bot, trigger):
     it will complain. If nothing can be found, it'll complain that the argument
     is not a valid timezone.
 
-    .. seealso::
-
-        Function :func:`~sopel.tools.time.format_time` is used to format
-        the current datetime according to the timezone (if found).
-
+    The format used can be set with the ``.settf`` or ``.setctf`` commands.
```
Comment (dgw) [inline sopel/modules/clock.py:37]: So, in a PR to add missing help prefixes, we're going to create places where the help prefix can't be used? AFAIK there's no way to replace these `.`s with the correct prefix at runtime, unless you've got some fancy trick up your sleeve in that sopel-help plugin. 😛

Comment (Exirel) [reply sopel/modules/clock.py:37]: Hm. Maybe we should just not use any prefix for command name? After all, user should understand what it means, right?

Comment (dgw) [reply sopel/modules/clock.py:37]: Should, yes. Doesn't mean we shouldn't try to develop some way of marking up a command name in command docstrings so the bot can add the prefix, but that sounds like a future feature thing.

Comment (Exirel) [reply sopel/modules/clock.py:37]: `*scream in refactoring*`

File: sopel/modules/clock.py [hunk 3]
```diff
@@ -299,20 +309,24 @@ def update_channel_format(bot, trigger):
...
+@module.example('.getctf #sopel', user_help=True)
+@module.example('.getctf', user_help=True)
 def get_channel_format(bot, trigger):
-    """
-    Gets the channel's preferred time format; will return current channel's if
-    no channel name is given.
-    """
+    """Gets the channel's preferred time format
 
+    It returns current channel's if no channel name is given.
```
Comment (dgw) [inline sopel/modules/clock.py:328]:
```suggestion
    It returns the current channel's if no channel name is given.
```

Did I miss this earlier? Damn.

**Pull request merged**